sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -158,37 +163,63 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
     private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
         CosmosEndToEndOperationLatencyPolicyConfig config = request.requestContext.getEndToEndOperationLatencyPolicyConfig();
         AvailabilityStrategy strategy = config.getAvailabilityStrategy();
-        List<Mono<StoreResponse>> monoList = new ArrayList<>();
-        List<RxDocumentServiceRequest> requestList = new ArrayList<>();
 
-        if (strategy instanceof ThresholdBasedAvailabilityStrategy) {
-            List<URI> effectiveEndpoints = getApplicableEndPoints(request);
-            if (effectiveEndpoints != null) {
-                effectiveEndpoints
-                    .forEach(locationURI -> {
-                        if (locationURI != null) {
-                            RxDocumentServiceRequest newRequest = request.clone();
-                            newRequest.requestContext.routeToLocation(locationURI);
-                            requestList.add(newRequest);
-                            if (monoList.isEmpty()) {
-                                monoList.add(getStoreResponseMono(newRequest, forceRefreshAndTimeout));
-                            } else {
-                                monoList.add(getStoreResponseMono(newRequest, forceRefreshAndTimeout)
-                                    .delaySubscription(((ThresholdBasedAvailabilityStrategy) strategy).getThreshold()
-                                        .plus(((ThresholdBasedAvailabilityStrategy) strategy)
-                                            .getThresholdStep().multipliedBy(monoList.size() - 1))));
-                            }
-                        }
-                    });
-            }
+        if (!(strategy instanceof ThresholdBasedAvailabilityStrategy)) {
+            // If no threshold based availability strategy is defined just return original Mono
+            return getStoreResponseMono(request, forceRefreshAndTimeout);
         }
 
-        // If the above conditions are not met, then we will just return the original request
-        if (monoList.isEmpty()) {
-            monoList.add(getStoreResponseMono(request, forceRefreshAndTimeout));
+        List<Mono<StoreResponse>> monoList = new ArrayList<>();
+        List<RxDocumentServiceRequest> requestList = new ArrayList<>();
+        final List<URI> effectiveEndpoints = getApplicableEndPoints(request);
+        if (effectiveEndpoints == null || effectiveEndpoints.stream().noneMatch(locationURI -> locationURI != null)) {
```

Comment:
nit: though this check is good to have, but we are streaming on the `effectiveEndpoints` twice for each request, one time here, and one below where we loop over them. Will it add any extra overhead? Hopefully not.